### ISSUE TITLE
When, if ever, will PlayCanvas Engine implement asm.js ?

### ISSUE THREAD

**Issue by SirFizX** [playcanvas/engine#369]
I hear that webGL, webAudio, and web video will soon be able to run on worker threads.  Once this is in place, it would be nice to have:
1) multiple canvas layers running in separate worker threads with precompiled javascript via [asm.js](http://asmjs.org/)
2) Audio/video streams running unblocked in worker threads with precompiled javascript via [asm.js](http://asmjs.org/)
3) only game set-up, logic and user io via main thread

**Comment by willeastcott** [comment]
You can already run PlayCanvas in a worker. See issue https://github.com/playcanvas/engine/issues/11.

**Comment by Guam** [comment]
Thanks Will.  That issue thread is very helpful.
